README.md

```diff
@@ -1,24 +1,79 @@
 # registered-email-address-web
-Web front-end for the Registered Email Address service
 
-### Requirements
+Web front-end for the Registered Email Address service. Provides the ability to change
+the registered email address for a company.
+
+## Requirements
 
 In order to run the service locally you will need the following:
 
 - [NodeJS](https://nodejs.org/en/)
 - [Git](https://git-scm.com/downloads)
 
+## Environment Variables
+
+By default will assume running within the local Docker environment.
+
+- `ACCOUNT_URL=<url>` - URL to the account service
+- `API_URL=<url>` - Url to public apis
+- `CACHE_SERVER=<host:port>` - hostname/port reference to the Cache server
+- `CDN_HOST=<host>` - Host url for the root of the CDN
+- `CDN_URL_CSS=<url>` - Path from the CDN URL to the CSS assets
+- `CDN_URL_JS=<url>` - Path from the CDN URL to the Javascript assets
+- `CHS_API_KEY=<apikey>` - API Key to use with CHS Services against public APIs
+- `CHS_INTERNAL_API_KEY=<apikey>` - API Key to use with CHS Services against private APIs
+- `CHS_URL=<url>` - URL for chs
+- `COOKIE_DOMAIN=<domain>` - Domain for which the cookie is set for
+- `COOKIE_NAME=<name>` - Name of the cookie set in users browser
+- `COOKIE_SECRET=<secret>` - the Secret Key to use to encrypt cookies/session data
+- `DEFAULT_SESSION_EXPIRATION=<expirationseconds>` - Time for the session to live for until expiry
+- `LOG_LEVEL=<level>` - Level Of Logging - DEBUG/INFO/WARN/ERROR
+- `NODE_ENV=<env name>` - Name of the environment being run within
+- `ORACLE_QUERY_API_URL=<url>` - URL to the oracle-query-api servicd
+- `PIWIK_START_GOAL_ID=<id>` - Goal ID
+- `PIWIK_SITE_ID=<id>` - Site ID to be analysed
+- `PIWIK_URL=<url>` - URL to analytics host
+- `PORT=<number>` - Port number to run the application on
+- `SESSION_COUNTDOWN=<length of countdown in seconds>` - timeout popup before signing out - default if not set=60 seconds
+- `SESSION_TIMEOUT=<browser timeout in seconds>` - default if not set=3600
+- `WEBFILING_LOGIN_URL=<url>` - URL to the web-filing login
+
+## Summary of User Journey
+
+1. User initiates email change
+2. If the user is not currently logged in the user is redirected to login
+3. Lookup company by company number
+4. Confirm the company choice
+5. If the user has not authenticated against the business with the company's
+  auth code, they are prompted to provide it
+6. Provide new email address
+7. Check answers
+8. User submits and sees the Update submitted message
+
+## Notable Third Party Frameworks used
+
+- `express` for http server/routing
+- `joi` for body validation
+
+  If modifying the body being posted to `/change-email-address` the `joi`
+  schema **must** be updated otherwise requests will fail. and the errors may
+  be quite opaque.
+- `govuk-frontend` etc. for providing Nunjucks components used to compose
+  the screens.
+
+## Developing the service locally
+
```

Comment:
Prerequisites: a populated CHIPS database. You will need to know the user schema for this - it takes the form `CAPDEVXX2` for example `CAPDEVJS2`

If you have not already got such a populated CHIPS db, you will need to request a schema. See https://github.com/companieshouse/docker-chs-development/blob/master/docs/chips.md